README.md

```diff
@@ -123,6 +123,14 @@ To manually run RCCL with NPKit enabled, environment variable `NPKIT_DUMP_DIR` n
 
 To manually analyze NPKit dump results, please leverage [npkit_trace_generator.py](https://github.com/microsoft/NPKit/blob/main/rccl_samples/npkit_trace_generator.py).
 
+## MSCCL/MSCCL++
+RCCL integrates MSCCL(https://github.com/microsoft/msccl) and MSCCL++ (https://github.com/microsoft/mscclpp) to leverage the highly efficient GPU-GPU communication primitives for collective operations. Thanks to Microsoft Corporation for collaborating with us in this project.
+
+MSCCL uses XMLs for different collective algorithms on different architectures. RCCL collectives can leverage those algorithms once the corresponding XML has been provided by the user. The XML files contain the sequence of send-recv and reduction operations to be executed by the kernel. On MI300X, MSCCL is enabled by default. On other platforms, the users may have to enable this by setting `RCCL_MSCCL_FORCE_ENABLE=1`.
+
+On the other hand, RCCL can be built with MSCCL++ support by providing the `-DENABLE_MSCCLPP=ON` flag during compilation via `cmake`. When using the `install.sh` script, `--enable-mscclpp` flag should be used. RCCL allreduce and allgather collectives can leverage the efficient MSCCL++ communication kernels for certain message sizes. Users need to set the RCCL environment variable `RCCL_ENABLE_MSCCLPP=1` to run RCCL workload with MSCCL++ support. It is also possible to set the message size threshold for using MSCCL++ by using the environment variable `RCCL_MSCCLPP_THRESHOLD`. Once `RCCL_MSCCLPP_THRESHOLD` is set, RCCL will invoke MSCCL++ kernels for all message sizes less than or equal to the specified threshold.
```

Comment:
Maybe consider documenting the default threshold value, but that might not be necessary.